### ISSUE TITLE
"Slide in" menus are not keyboard accessible

### ISSUE THREAD

**Issue by aitchiss** [forem/forem#14839]
**Describe the bug**

There are a few places in the app where we show a "hamburger" menu to open more details. For example:

- Left side of the header when in a mobile viewport
- Left and right side of a tag page (e.g.https://dev.to/t/javascript) when in tablet viewport or smaller

These menus currently lack the ARIA attributes to convey they are popup views, and the focus management required to reach and operate them properly by keyboard. For example, if you open the menu with the keyboard, then keep pressing tab, you'll notice that your keyboard focus continues to scroll behind the menu content and it's actually very hard to reach any of the links in the menu.

Although these views are primarily visible when using a touch device, they need to be keyboard accessible because:

- Some users will utilise keyboards, switch controls, or other assistive technology with a touch device
- Screen reader users rely on keyboard focus being in the correct place when dialogs open/close, even when swipe gestures are used on mobile devices
- Desktop users may see these layouts if zooming in to view the page more clearly

**To Reproduce**

- Using your devtools, or by resizing/zooming the window, shrink the viewport so that the hamburger menu is visible 
- Using the keyboard, Tab to the hamburger menu button and press Enter
- Keep tabbing and notice it's really hard to reach the menu content and you're scrolling behind the menu

**Expected behavior**

I think the most straightforward approach will be to treat these menus like our dropdowns. We should be able to re-use our `initializeDropdown` helper function to get all these behaviors without much additional code, but in summary:

- The "hamburger button" has the ARIA attributes `aria-haspopup`, `aria-controls` and `aria-expanded`, so that screen reader users know that it activates some dynamic content
- When open `aria-expanded` is "true"
- When closed, `aria-expanded` is "false"
- When the menu opens, focus is sent to the first link inside it
- When open, the `Escape` key closes the menu
- When closed either by `Escape` or by clicking the 'X' button, focus is returned to the "hamburger button"
- If the user Tabs past the last link in the menu, the menu closes

**Screenshots**

https://user-images.githubusercontent.com/20773163/135097487-b6a5d4d6-a9bc-48e6-a08a-67445e11443d.mp4

**Comment by Sara-Harvey** [comment]
Hey! If this is available, I'd like to tackle it.

**Comment by cmgorton** [comment]
Thanks @Sara-Harvey. I assigned it to you. Let us know if you have any questions.

**Comment by Sara-Harvey** [comment]
Hi-- I forked/cloned forem locally (Mac OS), and think I followed all installation directions. On setup, though, I get "URI::BadURIError: both URI are relative /Users/(myname)/Desktop/fp/forem/app/lib/url.rb:19:in `url'" I looked into fixes, but the only thing that works so far is to replace url.rb line 19:
base_url = "#{protocol}#{domain}" with
base_url = "http://#{domain}"
Not sure that's advisable.

I'm an open source newbie-- should the above be a new bug issue? Should I try to fix? How to proceed? 
I also installed the necessary programs, and am running Postgres and Redis, but should I have opened forem in Docker? I'm in the Dockerfile, and can't grok how I'd use Docker's sample command: docker start my_container
So, haven't been able to run locally yet. Thanks!
